### ISSUE TITLE
FanControl Updater.exe is being flagged by Defender as a Trojan Upon Executing the Updater in the App

### ISSUE THREAD

**Issue by abqnm** [Rem0o/FanControl.Releases#1294]
When attempting to update FanControl from v136 to v137, upon clicking update in the app, Defender immediately blocks the execution of updater.exe as being a trojan "Trojan:Win32/Wacatac.B!ml" and categorizes it as a "severe" threat.  I have never had this happen on previous updates, and was wondering if something with the updater changed, or if this is likely now just a false positive.  It still updated when I ran it the third time, even though Defender still said it blocked it.  Not sure what to take from this, but figured I should report it just in case the updater was compromised server-side or something...

![FanControl Updater Trojan Warning](https://user-images.githubusercontent.com/9168981/202870431-599a321d-8d3e-4af4-8259-0d11ac09868f.png)

**Comment by abqnm** [comment]
To add, I just saw that the updater is a new binary as of an hour ago.  I tried downloading the raw updater binary directly from github, and was also blocked by Defender with the same trojan warning.

**Comment by Mersuri** [comment]
Trojan:Win32/Bearfoos.A!ml for me.

**Comment by brunoherrera** [comment]
also it now says Version: 137 for me, not sure if it did update before detecting it or what

**Comment by brunoherrera** [comment]
> Trojan:Win32/Bearfoos.A!ml for me.

same here

**Comment by abqnm** [comment]
Yeah mine did update the app still on the third attempt at running it,
despite Defender still telling me it was blocked. I just don't think it
responded fast enough that time, and since the actual update is so fast, it
beat Defender out.

Interesting that Defender is flagging it as a different Trojan on your
system.

**Comment by machineglow** [comment]
same:  

Trojan:Win32/Bearfoos.A!ml
file: C:\Users\<username>\Documents\FanControl\Updater.exe

**Comment by Rem0o** [comment]
Sent for submission to Microsoft Defender. I don't know why everytime I update the binaries for the updater it gets flagged.

**Comment by brunoherrera** [comment]
> Sent for submission to Microsoft Defender. I don't know why everytime I update the binaries for the updater it gets flagged.


for me it was the first time ever, and it was an automatic update, when you get the little prompt and click udpate or ignore

**Comment by abqnm** [comment]
> Sent for submission to Microsoft Defender. I don't know why everytime I update the binaries for the updater it gets flagged.
> 
> For pure transparency purposes, here are the changes:
> 
> * Prepare the ground to be net48 and net7 compatible.
> * Call the version.json on this github to know which version is being downloaded, and display it in the console.
> * Remove the bitly link that was used to track the update count.
> * Replaced it with an google analytic event tag, same as https://getfancontrol.com/.

Thanks, I figured it was something along these lines. I'm surprised to hear that you've encountered this before, as I've never encountered it, and I've been using this app since the very, very early versions. I guess I must have always downloaded the update after you had submitted to MS for Defender whitelisting.

Regardless, I appreciate the explanation and I will leave this open until MS whitelists the new updater binary so that nobody else starts a new issue thread in the mean time.

**Comment by abqnm** [comment]
Updated my Defender definitions file, and it is already whitelisted.  Closing issue as solved.
